src/pages/fieldManagement/settings/DefaultFieldTable.jsx

```diff
@@ -115,22 +115,30 @@ export default function DefaultFieldTable({
             onCloseEditor();
           }}
           onSubmit={() => {
-            if (editField.id === 'region') {
+            if (editField?.id === 'region') {
               putSiteSetting(
                 editField.backendPath,
                 formSettings.regions,
               ).then(success => {
                 if (success) onCloseEditor();
               });
             }
-            if (editField.id === 'species') {
+            if (editField?.id === 'species') {
               putSiteSetting(
                 editField.backendPath,
                 formSettings.species,
               ).then(success => {
                 if (success) onCloseEditor();
               });
             }
+            if (editField?.id === 'relationship') {
+              putSiteSetting(
+                editField.backendPath,
+                formSettings.relationships,
+              ).then(success => {
```

Comment:
do you mind changing these to `async/await`? i want to use the same asynchronous patterns everywhere and i dunno why i deviated here.

Reply:
also it does seem like we could DRY this whole block up a little...

```
const relevantFormSettingsProperty = ... // you would set this part up and give it a better name 
const requestSuccessful = await putSiteSetting(
    editField.backendPath,
    relevantFormSettingsProperty,
);
if (requestSuccessful) onCloseEditor();
```

Reply:
Actually, this ended up being where I handled relationship deduplication. So, it's not as repetitive as it was before.